packages/workspace/src/utilities/typescript/ts-solution-setup.ts

```diff
@@ -75,3 +77,47 @@ export function isUsingTsSolutionSetup(tree?: Tree): boolean {
     isWorkspaceSetupWithTsSolution(tree)
   );
 }
+
+export function addProjectToTsSolutionWorkspace(
+  tree: Tree,
+  projectDir: string
+) {
+  const isIncluded = isProjectIncludedInPackageManagerWorkspaces(
+    tree,
+    projectDir
+  );
+  if (isIncluded) {
+    return;
+  }
+
+  // If dir is "libs/foo" then use "libs/*" so we don't need so many entries in the workspace file.
```

Comment:
Do we really need to run the risk of side-effects at all?

Why not just go for the project's specific location and leave it at that? Folks can always adjust and generalize if they want to, but this way Nx will always be correct and the move will be side-effect free for other packages?

Reply:
This makes sense. This file is actually a mirror of another helper in the JS package (to prevent circular deps). I will fix this in both places and will explore adding the pattern only if it doesn't pick up any extra projects. Otherwise, we explicitly add the project root. That way, we avoid including already existing projects that were not previously matched, while still being able to have more generic patterns to not pollute this file with a lot of entries.